### PR TITLE
Registry creation

### PR DESCRIPTION
**Pull request by strakam**
Gentlemen,

I just wanted to tell you whats new:

- Envs are now created via `gym.make` and users can create their own defaults via `gym.register`. Gym make is nice because it checks whether environment is working as it should (for example it caught that `.reset()` with fixed seed gave two different observations. 
- New `AgentFactory` that will create agents
- SB3 checks are subset of `gym.check_env`, so I removed them

This message is more important than the code, you don't need to look at the code changes.

Hopefully there will be only one more pull request before `v1.0` -- the one showcasing how to use some RL framework for this game.

Cheers 🚀

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,27 +45,30 @@ pip install -e .
...
+    action = agent.act(observation)
     observation, reward, terminated, truncated, info = env.step(action)
-    done = terminated or truncated
-    env.render(fps=6)
+    env.render()
 ```
-You can also check an example for 🦁[PettingZoo](./examples/pettingzoo_example.py) or more extensive
-example [here](./examples/complete_example.py).
+You can also check an example for 🦁[PettingZoo](./examples/pettingzoo_example.py) or a commented
+example showcasing features [here](./examples/complete_example.py).
```
Comment (jdujava) [inline README.md:71]: Something like "..an example with commentary showcasing various features.." sounds a little bit nicer IMO.

Comment (strakam) [reply README.md:71]: Indeed, will edit.

File: generals/agents/agent_factory.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+class AgentFactory:
+    """
+    Factory class for creating agents.
+    """
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def make_agent(agent_type: str, **kwargs) -> Agent:
```
Comment (Puckoland) [inline generals/agents/agent_factory.py:14]: It might be nice to name it simply `make`. We know that it will make an `agent` since it is `AgentFactory`.

Comment (strakam) [reply generals/agents/agent_factory.py:14]: true, will do in later updates, thanks

**Pull request merged**